### PR TITLE
Feature/speaker management

### PR DESCRIPTION
**Pull request by raghothams**
Feature to handle speaker management

### CODE REVIEW

File: guby_backend/views.py [hunk 1]
```diff
@@ -173,3 +173,45 @@ def event_delete(request, meetup_id, event_id):
...
+@has_ownership
+def speaker_add(request, meetup_id, event_id):
+    event = get_object_or_404(Event, pk=event_id)
+
+    if request.method == 'POST':
+        email = request.POST['speaker-email']
+        speaker_user = User.objects.filter(email=email)
+
+        # if username does not exists, create an inactive user and add them to speaker group
+        if speaker_user is None or len(speaker_user) == 0:
```
Comment (shrayasr) [inline guby_backend/views.py:187]: I don't like this checking for length :/ 

But the alternative is

```py
User.objects.get(email=email)
```

but this throws an exception if a user doesn't exist 😒

**Comment by shrayasr** [comment]
+1 from me otherwise. @nischalhp

File: guby_backend/views.py (hunk 1 above)
Comment (raghothams) [reply guby_backend/views.py:187]: Yes, I did try `get` initially, but that throws an exception

File: guby_backend/urls.py [hunk 2]
```diff
@@ -20,4 +20,5 @@
     path('meetups/<int:meetup_id>/events/<int:event_id>/speakers/', views.speaker_index, name='speaker_index'),
     path('meetups/<int:meetup_id>/events/<int:event_id>/speakers/add/', views.speaker_add, name='speaker_add'),
     path('meetups/<int:meetup_id>/events/<int:event_id>/speakers/<int:user_id>/delete/', views.speaker_delete, name='speaker_delete'),
+    path('meetups/<int:meetup_id>/events/<int:event_id>/speakers/<email>/delete/', views.speaker_delete_inactive, name='speaker_delete_inactive'),
```
[the remark this answers is not in the document]
Comment (raghothams) [reply guby_backend/urls.py:23]: That was my first try as well. But for some reason, it does not accept `<string:email>`. Throws an error

[the remark this answers is not in the document]
Comment (shrayasr) [reply guby_backend/urls.py:23]: Maybe everything is string by default. 

Oh well 🤷‍♀️ then +1 from me.

File: guby_backend/signals.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from django.contrib.auth.models import Group
+from django.contrib.auth import get_user_model
+from guby_backend.models import InactiveSpeaker, Event
+
+User = get_user_model()
+
+@receiver(post_save, sender=User)
```
Comment (nischalhp) [inline guby_backend/signals.py:9]: This decorator is so cool.

**Pull request merged**